rtCommon/bidsInterface.py

```diff
@@ -1,43 +1,292 @@
+        self.filename = self.niftiImage.get_filename()
+        self.imgVolumes = nib.four_to_three(self.niftiImage)
+        self.metadata = self.bidsArchive.getSidecarMetadata(self.filename, includeEntities=True)
+        self.metadata.pop('extension')
+        self.numVolumes = len(self.imgVolumes)
+        self.nextVol = 0
+
+    def getNumVolumes(self) -> int:
+        """Return the number of brain volumes in the run"""
+        # TODO - when we have BidsRun
+        # return self.bidsRun.getNumVolumes()
+        return self.numVolumes
+
+    def getIncremental(self, volIdx) -> BidsIncremental:
+        """
+        Get a BIDS incremental for the indicated index in the current subject/run
+        Args:
+            volIdx: The volume index (or TR) within the run to retrieve.
+        Returns:
+            BidsIncremental of that volume index within this subject/run
+        """
+        # TODO - when we have BidsRun
+        # return self.bidsRun.getIncremental(volIdx)
+        if volIdx < 0:
+            volIdx = self.nextVol
+        if volIdx < self.numVolumes:
+            self.nextVol += 1
+            return BidsIncremental(self.imgVolumes[volIdx], self.metadata)
+        else:
+            return None
+
+
+class OpenNeuroStream(BidsStream):
+    """
+    A BidsStream from an OpenNeuro dataset. The OpenNeuro dataset will be automatically
+    downloaded at the remote as needed to satisfy the stream of data.
+    """
+    def __init__(self, dsAccessionNumber, **entities):
+        """
+        Args:
+            dsAccessionNumber: The OpenNeruo specific accession number for the dataset
+                to stream.
+            entities: BIDS entities (subject, session, task, run, suffix, datatype) that
+                define the particular subject/run of the data to stream
+        """
+        subject = entities.get('subject')
+        run = entities.get('run')
+        if subject is None or run is None:
+            raise RequestError("OpenNeuroStream: Must specify subject and run number")
+        # TODO - Use OpenNeuroService when it is available, to download
+        #   and access the dataset and get dataset entities
+        # OpenNeuroService to provide path to dataset
+        datasetPath = tmpDownloadOpenNeuro(dsAccessionNumber, subject, run)
+        super().__init__(datasetPath, **entities)
+
+
+def tmpDownloadOpenNeuro(dsAccessNumber, subject, run) -> str:
+    """
+    Temporary function used until we integrate in the OpenNeuro service. Downloads
+    a portion of an OpenNeuro dataset corresponding to the subject/run.
+    Args:
+        dsAccessionNumber: The OpenNeruo specific accession number for the dataset
+            to stream.
+        subject: the specific subject name within the OpenNeuro dataset to download
+        run: the specific run within the subject's data to download.
+    Returns:
+        Absolute path to where the dataset has been downloaded.
+
+    """
+    tmpDir = '/tmp/openneuro'
```

Comment:
Might be better to use `tempfile` interface here to be cross-platform compatible